src/contexts/InjectedProviderContext.jsx

```diff
@@ -128,6 +128,11 @@ export const InjectedProvider = ({ children }) => {
     setAddress(null);
     setWeb3Modal(defaultModal);
     web3Modal.clearCachedProvider();
+    if (localStorage.getItem('walletconnect')) {
```

Comment:
Found that this is required, otherwise web3Modal doesn't allow you to choose another wallet provider after you choose WalletConnect